contracts/src/PDPService.sol

```diff
@@ -80,6 +85,8 @@ contract PDPService {
     mapping(uint256 => uint256) nextChallengeEpoch;
     // Each proof set notifies a configurable listener to implement extensible applications managing data storage
     mapping(uint256 => address) proofSetListener;
+    mapping(uint256 => uint256) lastChallengedLeaf;
```

Comment:
Nit: I think we can improve this name. On the surface it sounds like the index of the leaf that was most recently actually challenged (i.e. after sampling randomness and performing modulus). The "last" and past tense here conflicts with "next" challenge epoch, where they should be the same thing (I know one is time vs one is space). Suggestions.
- challengeRange
- challengeLeafCount

In all cases it probably deserves a comment.